packages/compile-solidity/legacy/shims.js

```diff
@@ -1,52 +1,15 @@
-function shimOutput({ contracts: list, sourceIndexes, compilerInfo }) {
+const { Shims } = require("@truffle/compile-common");
+
+function shimOutput({ contracts: list, sourceIndexes, compiler }) {
   const contracts = list
     // get old format
-    .map(contract => shimContract(contract))
+    .map(contract => Shims.NewToLegacy.forContract(contract))
     // get pair
     .map(contract => ({ [contract.contract_name]: contract }))
     // merge pairs
     .reduce((a, b) => Object.assign({}, a, b), {});
 
-  return [contracts, sourceIndexes, compilerInfo];
-}
-
-function shimContract(contract) {
-  const {
-    contractName,
-    sourcePath,
-    source,
-    sourceMap,
-    deployedSourceMap,
-    legacyAST,
-    ast,
-    abi,
-    metadata,
-    bytecode,
-    deployedBytecode,
-    compiler,
-    devdoc,
-    userdoc,
-    immutableReferences
-  } = contract;
-
-  return {
-    contract_name: contractName,
-    sourcePath,
-    source,
-    sourceMap,
-    deployedSourceMap,
-    legacyAST,
-    ast,
-    abi,
-    metadata,
-    bytecode: shimBytecode(bytecode),
-    deployedBytecode: shimBytecode(deployedBytecode),
-    unlinked_binary: shimBytecode(bytecode),
-    compiler,
-    devdoc,
-    userdoc,
-    immutableReferences
-  };
+  return [contracts, sourceIndexes, compiler];
 }
 
 function shimBytecode(bytecode) {
```

Comment:
We're not using this one, either, right?

Reply:
Right you are again.